### ISSUE TITLE
Failed FileVsBufferTest

### ISSUE THREAD

**Issue by tc64** [assaft/caevo#5]
Hi Assaft,

I checked out your branch and was able to compile but failed 2 tests. One of them is the FileVsBufferTest. Specifically, testRawToTimex fails. Here is the stack trace - any idea what's going on?

at junit.framework.Assert.assertEquals(Assert.java:81)
        at caevo.FileVsBufferTest.testFileVsBuffer(FileVsBufferTest.java:43)
        at caevo.FileVsBufferTest.testRawToTimex(FileVsBufferTest.java:54)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at junit.framework.TestCase.runTest(TestCase.java:168)
        at junit.framework.TestCase.runBare(TestCase.java:134)
        at junit.framework.TestResult$1.protect(TestResult.java:110)
        at junit.framework.TestResult.runProtected(TestResult.java:128)
        at junit.framework.TestResult.run(TestResult.java:113)
        at junit.framework.TestCase.run(TestCase.java:124)
        at junit.framework.TestSuite.runTest(TestSuite.java:232)
        at junit.framework.TestSuite.run(TestSuite.java:227)
        at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:83)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
        at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
        at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)

**Comment by assaft** [comment]
Hi tc! 
I'm trying to reproduce the problem but it works fine on my side. Could you please include the console prints?
Thanks!

**Comment by tc64** [comment]
Thanks for the quick reply! Here is the entire console output from the surefire report - is this sufficient?

---

## Test set: caevo.FileVsBufferTest

Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 4.469 sec <<< FAILURE!
testRawToTimex(caevo.FileVsBufferTest)  Time elapsed: 4.27 sec  <<< FAILURE!
junit.framework.ComparisonFailure: Generated XML. expected:<..." encoding="UTF-8"?>[
<root xmlns="http://chambers.com/corpusinfo">
  <file name="input.txt">
    <entry sid="0" file="input.txt">
      <sentence>Libya brought the case in 2003 to Britain because of 11/25/1980 and complained .</sentence>
      <tokens>
        <t>"" "Libya" " "</t>
        <t>" " "brought" " "</t>
        <t>" " "the" " "</t>
        <t>" " "case" " "</t>
        <t>" " "in" " "</t>
        <t>" " "2003" " "</t>
        <t>" " "to" " "</t>
        <t>" " "Britain" " "</t>
        <t>" " "because" " "</t>
        <t>" " "of" " "</t>
        <t>" " "11/25/1980" " "</t>
        <t>" " "and" " "</t>
        <t>" " "complained" ""</t>
        <t>"" "." ""</t>
      </tokens>
      <parse>(ROOT (S (NP (NNP Libya)) (VP (VP (VBD brought) (NP (DT the) (NN case)) (PP (IN in) (NP (CD 2003))) (PP (TO to) (NP (NP (NNP Britain)) (PP (RB because) (IN of) (NP (CD 11/25/1980)))))) (CC and) (VP (VBD complained))) (. .)))</parse>
      <deps>nsubj(brought-2, Libya-1)
nsubj(complained-13, Libya-1)
root(ROOT-0, brought-2)
det(case-4, the-3)
dobj(brought-2, case-4)
case(2003-6, in-5)
nmod:in(brought-2, 2003-6)
case(Britain-8, to-7)
nmod:to(brought-2, Britain-8)
case(11/25/1980-11, because-9)
mwe(because-9, of-10)
nmod:because_of(Britain-8, 11/25/1980-11)
cc(brought-2, and-12)
conj:and(brought-2, complained-13)</deps>
      <events>
        <event id="e1" eiid="ei1" offset="2" string="brought" tense="PAST" aspect="NONE" class="OCCURRENCE" polarity="POS" modality="" happen="" lowerBoundDuration="" upperBoundDuration="" />
        <event id="e2" eiid="ei2" offset="13" string="complained" tense="PAST" aspect="NONE" class="OCCURRENCE" polarity="POS" modality="" happen="" lowerBoundDuration="" upperBoundDuration="" />
      </events>
      <timexes>
        <timex tid="t1" text="2003" offset="6" length="1" type="DATE" value="2003" temporalFunction="false" />
        <timex tid="t2" text="11/25/1980" offset="11" length="1" type="DATE" value="1980-11-25" temporalFunction="false" />
      </timexes>
    </entry>
    <tlink event1="t1" event2="t2" relation="AFTER" closed="false" origin="TimeTimeSieve" type="tt" />
    <tlink event1="ei1" event2="t1" relation="IS_INCLUDED" closed="false" origin="MLEventTimeSameSent" type="et" />
    <tlink event1="ei2" event2="t1" relation="IS_INCLUDED" closed="false" origin="MLEventTimeSameSent" type="et" />
    <tlink event1="ei1" event2="t2" relation="IS_INCLUDED" closed="false" origin="MLEventTimeSameSent" type="et" />
    <tlink event1="ei2" event2="t2" relation="IS_INCLUDED" closed="false" origin="MLEventTimeSameSent" type="et" />
    <tlink event1="ei1" event2="ei2" relation="VAGUE" closed="false" origin="AllVagueSieve" type="ee" />
  </file>
</root>
]

> but was:<..." encoding="UTF-8"?>[
> <root xmlns="http://chambers.com/corpusinfo">
>   <file name="input.txt">
>     <entry sid="0" file="input.txt">
>       <sentence>Libya brought the case in 2003 to Britain because of 11/25/1980 and complained .</sentence>
>       <tokens>
>         <t>"" "Libya" " "</t>
>         <t>" " "brought" " "</t>
>         <t>" " "the" " "</t>
>         <t>" " "case" " "</t>
>         <t>" " "in" " "</t>
>         <t>" " "2003" " "</t>
>         <t>" " "to" " "</t>
>         <t>" " "Britain" " "</t>
>         <t>" " "because" " "</t>
>         <t>" " "of" " "</t>
>         <t>" " "11/25/1980" " "</t>
>         <t>" " "and" " "</t>
>         <t>" " "complained" ""</t>
>         <t>"" "." ""</t>
>       </tokens>
>       <parse>(ROOT (S (NP (NNP Libya)) (VP (VP (VBD brought) (NP (DT the) (NN case)) (PP (IN in) (NP (CD 2003))) (PP (TO to) (NP (NP (NNP Britain)) (PP (RB because) (IN of) (NP (CD 11/25/1980)))))) (CC and) (VP (VBD complained))) (. .)))</parse>
>       <deps>nsubj(brought-2, Libya-1)
> nsubj(complained-13, Libya-1)
> root(ROOT-0, brought-2)
> det(case-4, the-3)
> dobj(brought-2, case-4)
> case(2003-6, in-5)
> nmod:in(brought-2, 2003-6)
> case(Britain-8, to-7)
> nmod:to(brought-2, Britain-8)
> case(11/25/1980-11, because-9)
> mwe(because-9, of-10)
> nmod:because_of(Britain-8, 11/25/1980-11)
> cc(brought-2, and-12)
> conj:and(brought-2, complained-13)</deps>
>       <events>
>         <event id="e1" eiid="ei1" offset="2" string="brought" tense="PAST" aspect="NONE" class="OCCURRENCE" polarity="POS" modality="" happen="" lowerBoundDuration="" upperBoundDuration="" />
>         <event id="e2" eiid="ei2" offset="13" string="complained" tense="PAST" aspect="NONE" class="OCCURRENCE" polarity="POS" modality="" happen="" lowerBoundDuration="" upperBoundDuration="" />
>       </events>
>       <timexes>
>         <timex tid="t1" text="2003" offset="6" length="1" type="DATE" value="2003" temporalFunction="false" />
>         <timex tid="t2" text="11/25/1980" offset="11" length="1" type="DATE" value="1980-11-25" temporalFunction="false" />
>       </timexes>
>     </entry>
>     <tlink event1="t1" event2="t2" relation="AFTER" closed="false" origin="TimeTimeSieve" type="tt" />
>     <tlink event1="ei1" event2="t1" relation="IS_INCLUDED" closed="false" origin="MLEventTimeSameSent" type="et" />
>     <tlink event1="ei2" event2="t1" relation="IS_INCLUDED" closed="false" origin="MLEventTimeSameSent" type="et" />
>     <tlink event1="ei1" event2="t2" relation="IS_INCLUDED" closed="false" origin="MLEventTimeSameSent" type="et" />
>     <tlink event1="ei2" event2="t2" relation="IS_INCLUDED" closed="false" origin="MLEventTimeSameSent" type="et" />
>     <tlink event1="ei1" event2="ei2" relation="VAGUE" closed="false" origin="AllVagueSieve" type="ee" />
>   </file>
> </root>

]

> ```
> at junit.framework.Assert.assertEquals(Assert.java:81)
> at caevo.FileVsBufferTest.testFileVsBuffer(FileVsBufferTest.java:43)
> at caevo.FileVsBufferTest.testRawToTimex(FileVsBufferTest.java:54)
> at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> at java.lang.reflect.Method.invoke(Method.java:497)
> at junit.framework.TestCase.runTest(TestCase.java:168)
> at junit.framework.TestCase.runBare(TestCase.java:134)
> at junit.framework.TestResult$1.protect(TestResult.java:110)
> at junit.framework.TestResult.runProtected(TestResult.java:128)
> at junit.framework.TestResult.run(TestResult.java:113)
> at junit.framework.TestCase.run(TestCase.java:124)
> at junit.framework.TestSuite.runTest(TestSuite.java:232)
> at junit.framework.TestSuite.run(TestSuite.java:227)
> at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:83)
> at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
> at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
> at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
> at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> at java.lang.reflect.Method.invoke(Method.java:497)
> at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
> at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
> at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
> at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
> at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
> ```

**Comment by assaft** [comment]
Thanks, they look quite the same except for the number of empty lines. I'm not sure why this happens. Can you attach the full console output that you get when running "mvn test" (it would be the best f you can attach it as a separate text file) ?

**Comment by tc64** [comment]
[console-out.txt](https://github.com/assaft/caevo/files/374285/console-out.txt)

**Comment by assaft** [comment]
Thanks, I looked at it but I haven't found the problem yet. The content of the buffer run and the file run look the same but probably there is some white space diff. I suspect the problem is related to Windows vs. Linux line ending convention (CRLF vs. LF) but I need to verify that on a Linux system. You can skip the test for the time being, as it may take a bit until I find the problem.

**Comment by tc64** [comment]
Thanks assaft. That explanation sonde plausible. I haven't tested on
Windows myself.

**Comment by assaft** [comment]
Fixed in commit e2e2626b0cf0127570bd5e5d0a181e021acef8d1

The problem was indeed inconsistent use of line ending conventions. For the output of the file annotation we used the operating system default line ending (\r\n on Windows, \n on Linux), while for the output of the buffer annotation we used \r\n (the default of Format.getPrettyXML(), see [here](http://www.jdom.org/docs/apidocs.1.1/org/jdom/output/Format.html#setLineSeparator%28java.lang.String%29)). As a result, the test passed on Windows and failed on Linux. The committed fix uses \r\n also for the output of the file annotation, ignoring the operating system.
